perception/autoware_traffic_light_selector/src/traffic_light_selector_utils.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "traffic_light_selector_utils.hpp"
+
+#include <algorithm>
+#include <vector>
+
+namespace autoware::traffic_light
+{
+namespace utils
+{
+
+bool isInsideRoughRoi(const RegionOfInterest & detected_roi, const RegionOfInterest & rough_roi)
+{
+  const auto tl_x = detected_roi.x_offset;
+  const auto tl_y = detected_roi.y_offset;
+  const auto br_x = detected_roi.x_offset + detected_roi.width;
+  const auto br_y = detected_roi.y_offset + detected_roi.height;
+
+  bool is_tl_inside = rough_roi.x_offset <= tl_x && tl_x <= rough_roi.x_offset + rough_roi.width &&
+                      rough_roi.y_offset <= tl_y && tl_y <= rough_roi.y_offset + rough_roi.height;
+  if (!is_tl_inside) {
+    return false;
+  }
+
+  bool is_br_inside = rough_roi.x_offset <= br_x && br_x <= rough_roi.x_offset + rough_roi.width &&
+                      rough_roi.y_offset <= br_y && br_y <= rough_roi.y_offset + rough_roi.height;
+  if (!is_br_inside) {
+    return false;
+  }
+
+  return true;
+}
+
+void computeCenterOffset(
+  const RegionOfInterest & source, const RegionOfInterest & target, int32_t & shift_x,
+  int32_t & shift_y)
+{
+  const auto source_roi_center_x = static_cast<int32_t>(source.x_offset + source.width / 2);
+  const auto source_roi_center_y = static_cast<int32_t>(source.y_offset + source.height / 2);
+  const auto target_roi_center_x = static_cast<int32_t>(target.x_offset + target.width / 2);
+  const auto target_roi_center_y = static_cast<int32_t>(target.y_offset + target.height / 2);
+  shift_x = source_roi_center_x - target_roi_center_x;
+  shift_y = source_roi_center_y - target_roi_center_y;
+}
+
+RegionOfInterest getShiftedRoi(
+  const RegionOfInterest & image, const uint32_t & image_width, const uint32_t & image_height,
```

Comment:
the parameter name `image` could cause confusion.
ROI or input_roi could be better?
```suggestion
  const RegionOfInterest & ROI, const uint32_t & image_width, const uint32_t & image_height,
```

Reply:
It is true, fixed in  [360782f](https://github.com/autowarefoundation/autoware_universe/pull/10352/commits/360782f85f0f1a03e24bdf712b08262733352916)